de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/StoppableDocumentListener.java

```diff
@@ -20,25 +22,55 @@
     private static final Logger LOG = Logger
         .getLogger(StoppableDocumentListener.class);
 
-    public StoppableDocumentListener(EditorManager editorManager) {
+    private final VirtualFileConverter virtualFileConverter;
+
+    public StoppableDocumentListener(EditorManager editorManager,
+        VirtualFileConverter virtualFileConverter) {
+
         super(editorManager);
         super.setEnabled(false);
+
+        this.virtualFileConverter = virtualFileConverter;
     }
 
     /**
-     * Calls
-     * {@link EditorManager#generateTextEdit(int, String, String, SPath)}
+     * Generates and dispatches a <code>TextEditActivity</code> for the given
+     * <code>DocumentEvent</code>.
      *
-     * @param event
+     * @param event the <code>DocumentEvent</code> to react to
      */
     @Override
     public void beforeDocumentChange(DocumentEvent event) {
-        // We rely on the editor pool to filter files that are not shared.
-        SPath path = editorManager.getEditorPool().getFile(event.getDocument());
+        Document document = event.getDocument();
+
+        SPath path = editorManager.getEditorPool().getFile(document);
+
         if (path == null) {
-            LOG.debug("Event for document " + event.getDocument()
-                    + " ignored: document is not known to the editor pool");
-            return;
+            VirtualFile virtualFile = FileDocumentManager.getInstance()
+                .getFile(document);
+
+            if (virtualFile == null) {
+                LOG.debug("Ignoring event for document " + document
```

Comment:
Are you tracking all files in the IDE ? If yes this could flood the log file a little bit depending on your sharing setup, e.g you do a partial sharing with only a few files.

Reply:
This patch does not increase the number of created log entries. Everything that is logged now would have just been logged beforehand as "... document is not known to the editor pool". But yes, you are right. This is tracking all files in the IDE. This will create a lot of logs, especially when working on non-shared files. I will set it to trace. (I have to modify this patch anyway as I can merge the last two if clauses.)